### PR TITLE
fix: populate prefixPath variable in views. Closes #119

### LINKED ISSUE #119: pathPrefix not getting set
**Describe the bug**

`pathPrefix` does not get set correctly in registration.hbs

**To Reproduce**

1. Set `BASE_URL=/anything`
2. Run server
3. Notice the handlebars templates don't respect the variable `pathPrefix`

**Expected behavior**

`pathPrefix` getting set to `BASE_URL` in the view file.

**Environment**

Docker Image

oryd/kratos-selfservice-ui-node:v0.6.0-alpha.2 

docker.io/oryd/kratos-selfservice-ui-node@sha256:255582ce33e6941374afb26cd11b2b78b32090e92164fa3065f592ba05e6e4d3

**Additional context**

BASE_URL is necessary, as ingress resources in kubernetes do not allow ports outside of 80 and 443.

### PR DESCRIPTION
**Pull request by sbussard**
This change is self-explanatory.

### CODE REVIEW

**Comment by sbussard** [comment]
Supporting this work
https://github.com/sbussard/kratos-kind/tree/main

File: src/index.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ app.set('view engine', 'hbs')
 app.use((req: Request, res: Response, next: NextFunction) => {
   res.locals.projectName = config.projectName
   res.locals.baseUrl = config.baseUrl
-  res.locals.pathPrefix = config.baseUrl ? '' : '/'
+  res.locals.pathPrefix = config.baseUrl ? config.baseUrl + '/' : ''
```
Comment (aeneasr) [inline src/index.ts:32]: This assumes that baseUrl always ends without trailing slash. Maybe use a path merge function instead?

**Review by aeneasr — CHANGES_REQUESTED** [review]
Awesome, thank you for your contribution! This looks pretty good and I have some ideas how to improve it further :)

File: src/index.ts (hunk 1 above)
Comment (sbussard) [reply src/index.ts]: Good call. I updated it to use `path.join`

**Comment by aeneasr** [comment]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

**Comment by harnash** [comment]
@sbussard + @aeneasr : I think this change broke the main layout.

Example HTML:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <base href="https://example.com/self-service/"/>
    <link rel="icon"
          type="image/png"
          href="https:/example.com/self-service/favico.png">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css">
    <meta charset="utf-8">
    <title>SecureApp</title>
    <link rel="stylesheet" href="https:/example.com/self-service/index.css">
    <link rel="stylesheet" href="https:/example.com/self-service/typography.css">
    <link rel="stylesheet" href="https:/example.com/self-service/form.css">
    <link rel="stylesheet" href="https:/example.com/self-service/auth.css">
</head>
<body>
```
see that the URLs get `https:/example.com` instead of `https://example.com`? Browser now tries to fetch it from the paths like this: `https://example.com/self-service/example.com/self-service/index.css`.

**Comment by aeneasr** [comment]
Yeah this expects the url to be a path. The right appraoch would probably be to strip the last `/` of `baseUrl` and then append it. Would you be up for a PR? :)

**Comment by harnash** [comment]
@aeneasr : here it is #120 I've tested it briefly and it seems to work just fine.

File: src/index.ts [hunk 2]
```diff
@@ -29,7 +30,7 @@ app.set('view engine', 'hbs')
 app.use((req: Request, res: Response, next: NextFunction) => {
   res.locals.projectName = config.projectName
   res.locals.baseUrl = config.baseUrl
-  res.locals.pathPrefix = config.baseUrl ? '' : '/'
+  res.locals.pathPrefix = path.join(config.baseUrl, './')
```
Comment (radekg) [inline src/index.ts:33]: This does not work @aeneasr and it is not self explanatory at all. What's the reason for this change?

```
$ node
require("path")
path.join("https://website.com/profile", "./")
```

produces

```
'https:/website.com/profile/'
```

verbatim. So this is terribly incorrect.

**Comment by sbussard** [comment]
@radekg this works for local relative paths. Not sure why absolute paths would be used here.
If that's a use-case you'd like to see supported, I'm sure the team would be open to a pull request. 

Please review this issue for more details
https://github.com/ory/kratos-selfservice-ui-node/issues/119

**Comment by radekg** [comment]
@sbussard it does not work because, currently, the `pathPrefix` is derived from `baseUrl` and a URL is not a path. It's been explained here: https://github.com/ory/kratos-selfservice-ui-node/pull/120#issuecomment-858806883 and fixed already in https://github.com/ory/kratos-selfservice-ui-node/pull/128.

**Comment by sbussard** [comment]
I think the original rationale was that web servers follow the Unix path convention (forward slash delimiters) so it made sense to me to use path.resolve. But it doesn't work for full urls, so that approach missed some important use cases. Glad it's fixed

**Pull request merged**